src/java/org/apache/cassandra/locator/Ec2MetadataServiceConnector.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.locator;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.utils.Clock;
+import org.apache.cassandra.utils.Pair;
+
+import static java.lang.String.format;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.joining;
+import static org.apache.cassandra.locator.Ec2MetadataServiceConnector.EC2MetadataType.v2;
+
+abstract class Ec2MetadataServiceConnector extends AbstractCloudMetadataServiceConnector
+{
+    static final String EC2_METADATA_TYPE_PROPERTY = "ec2_metadata_type";
+    static final String EC2_METADATA_URL_PROPERTY = "ec2_metadata_url";
+    static final String DEFAULT_EC2_METADATA_URL = "http://169.254.169.254";
+
+    Ec2MetadataServiceConnector(SnitchProperties properties)
+    {
+        metadataServiceUrl = properties.get(EC2_METADATA_URL_PROPERTY, DEFAULT_EC2_METADATA_URL);
+    }
+
+    enum EC2MetadataType
+    {
+        v1(props -> Ec2MetadataServiceConnector.V1Connector.create(props)),
```

Comment:
@jacek-lewandowski 

I had it like that but I liked this more because I do not need to use "toUpperCase()" when I am parsing it from the file. In the file, it will be probably "v1" or "v2" intead of "V1" or "V2". Also in 'toString' it would look strange to have it uppercased so I would have to lowercase it again. And all of this just to have an enum in uppercase ...